### PR TITLE
#39 integrate/verify login page

### LINKED ISSUE #39: Integrate/verify Login page works with backend
Fix login page so that tokens are stored and used correctly and redirects to browser page with correct location

### PR DESCRIPTION
**Pull request by FredSoderberg**
#### What does this PR do?
Verifies and implements backend integration into login.

##### Why are we doing this? Any context or related work?
minimum viable product implementation.

#### Where should a reviewer start?
Login.js check the willMount and then on componentDidUpdate

resolves #39

### CODE REVIEW

File: Bikeini/screens/Login.js [hunk 1]
```diff
@@ -53,72 +52,73 @@ const styles = StyleSheet.create({
...
+      !authState.loadingJwt
+      && authState.jwt[0]
+      && !profileState.profileLoaded
+      && !profileState.loadingProfile
+      && !profileState.error
+      && !authState.isLoggedIn
+      && !authState.deletingJwt) {
+      loadProfileInit(authState.jwt[0]);
+    } else if (profileState.profileLoaded && profileState.location.length) {
+      navigation.navigate('TempPage');
```
Comment (erikajansson1) [inline Bikeini/screens/Login.js]: Shouldn't this one lead to Browser?

File: Bikeini/utilities/serverApi.js [hunk 2]
```diff
@@ -2,23 +2,89 @@
...
         'Content-Type': _contentType,
         'x-access-token': _jwt,
       },
-    // Authorization: 'Bearer ' + 'SECRET_OAUTH2_TOKEN_IF_AUTH' },
     })
+      .then(handleErrors)
       .then(response => response.json());
   },
 
   get(_urlEnd, _jwt) {
```
Comment (erikajansson1) [inline Bikeini/utilities/serverApi.js:85]: Method named get, actual method used POST. A bit confusing, maybe change name?

File: Bikeini/utilities/deviceStorage.js [hunk 3]
```diff
@@ -16,16 +18,17 @@ const deviceStorage = {
 	    try {
 	      const value = await AsyncStorage.getItem('id_token');
 	      if (value !== null) {
-	      	console.log(value);
+        console.log('found:', value);
```
Comment (erikajansson1) [inline Bikeini/utilities/deviceStorage.js:21]: Is this one neccessary?

File: Bikeini/screens/Browser.js [hunk 4]
```diff
@@ -178,7 +178,7 @@ class Browser extends React.Component {
     const header = this.renderHeader();
     const filter = this.renderFilter();
     const list = this.renderList();
-
+    // console.log('profileState',this.props.profileState);
```
Comment (erikajansson1) [inline Bikeini/screens/Browser.js]: Unneccessary

File: Bikeini/navigation/actions/JwtActions.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  LOAD_JWT_FAILURE,
+  DELETE_JWT_BEGIN,
+  DELETE_JWT_SUCCES,
+  DELETE_JWT_FAILURE,
+  STORE_JWT_BEGIN,
+  STORE_JWT_FAILURE,
+  STORE_JWT_SUCCESS,
+} from './types';
+
+export const deleteJWTBegin = () => (
```
Comment (erikajansson1) [inline Bikeini/navigation/actions/JwtActions.js:14]: Good that this is in its own file.

**Comment by erikajansson1** [comment]
Great work! 🎉 Look at the comments and change what is necessary, otherwise good!

File: Bikeini/screens/Login.js (hunk 1 above)
Comment (FredSoderberg) [reply Bikeini/screens/Login.js]: fixed

File: Bikeini/utilities/serverApi.js (hunk 2 above)
Comment (FredSoderberg) [reply Bikeini/utilities/serverApi.js:85]: fixed, temp fix for previous bug which is not used anymore

File: Bikeini/screens/Browser.js (hunk 4 above)
Comment (FredSoderberg) [reply Bikeini/screens/Browser.js]: fixed

File: Bikeini/navigation/actions/JwtActions.js (hunk 5 above)
Comment (FredSoderberg) [reply Bikeini/navigation/actions/JwtActions.js:14]: thanks 👍 🐰

File: Bikeini/utilities/deviceStorage.js (hunk 3 above)
Comment (FredSoderberg) [reply Bikeini/utilities/deviceStorage.js:21]: nahh but it might be removed later, must verify its not in use anymore first

**Pull request merged**